### ISSUE TITLE
Output container bitrate values from `MetadataRetriever` (for parity with `android.media.MediaMetadataRetriever`)

### ISSUE THREAD

**Issue by besarmemishi** [google/ExoPlayer#10276]
## Question
I wonder why for some audio/video files (https sources) the bitrate returns -1, while the Android MetadataRetriever returns correct bitrate.

### Example
````kotlin

val trackGroupArray = MetadataRetriever.retrieveMetadata(
            context,
            mediaItem
).await()

  for (arrayIndex in 0 until trackGroupArray.length) {
            val trackGroup = trackGroupArray[arrayIndex]

            for (groupIndex in 0 until trackGroup.length) {
                val sampleMimeType = trackGroup.getFormat(groupIndex).sampleMimeType

                if (sampleMimeType?.contains("audio") == true) {
                    return trackGroup?.getFormat(0)?.bitrate   // bitrate -1
                }
            }
        }
````

### Version
` com.google.android.exoplayer:exoplayer:2.17.1`

**Comment by icbaker** [comment]
`-1` is [`Format.NO_VALUE`](https://exoplayer.dev/doc/reference/com/google/android/exoplayer2/Format.html#NO_VALUE) - so this suggests that ExoPlayer's `MetadataRetriever` didn't read a bitrate value from the provided media.

Have you read the caveats on [`Format.averageBitrate`](https://exoplayer.dev/doc/reference/com/google/android/exoplayer2/Format.html#averageBitrate) and [`Format.peakBitrate`](https://exoplayer.dev/doc/reference/com/google/android/exoplayer2/Format.html#peakBitrate) as to when you will see `NO_VALUE` (since `bitrate` is just one or other of these values)?

ExoPlayer metadata extraction is motivated by the information needed for playback. For non-adaptive playback (e.g. just an MP4 file) the bitrate isn't very useful, because there are no adaptive track selection decisions to be made - so it's often not read from these sorts of files.

> while the Android MetadataRetriever returns correct bitrate.

I assume you mean [`android.media.MediaMetadataRetriever`](https://developer.android.com/reference/android/media/MediaMetadataRetriever) here?

**Comment by icbaker** [comment]
I'm marking this as an enhancement to output these values from `MetadataRetriever`. Just to set expectations: we currently have no plans to work on this.

**Comment by besarmemishi** [comment]
Yes, I meant  `[android.media.MediaMetadataRetriever](https://developer.android.com/reference/android/media/MediaMetadataRetriever)``

I am using Exo MetadataRetriever as it is extremly faster comparing to MediaMetadataRetriever.

When I receive `-1` which happens for most of `.mp3` files, also the [Format.averageBitrate](https://exoplayer.dev/doc/reference/com/google/android/exoplayer2/Format.html#averageBitrate) and [Format.peakBitrate](https://exoplayer.dev/doc/reference/com/google/android/exoplayer2/Format.html#peakBitrate) are -1

Url: https://download.samplelib.com/mp3/sample-15s.mp3

![image](https://user-images.githubusercontent.com/105178492/169521980-3dd8c345-65b5-4af3-8ca5-f2e1d58c9e89.png)

**Comment by OxygenCobalt** [comment]
May be willing to work on this in the future if I get enough time. How feasible is it to get an approximate bitrate for VBR progressive containers @icbaker? One of the big reasons I want this to be added is so that I can show a correct approximate bitrate for VBR files, something that `MediaExtractor` doesn't do.

**Comment by Tolriq** [comment]
For the record Exoplayer already extract the mp3 bitrate, in ```mp3extractor.java``` `readInternal` you can just add ``` .setAverageBitrate(synchronizedHeader.bitrate)```.

Wanted to check for more formats and it's not that hard for Alac and FLAC (at least approximate).

I also wanted to expose bit per sample for FLAC but I'm not sure if adding a new field to MetaData would have been accepted.

**Comment by rohitjoins** [comment]
Hi @besarmemishi,

We have recently implemented changes to expose bitrate information from `MetadataRetriever` for progressive media formats. I'll close this issue now as this work is complete. Please feel free to reopen the issue if it doesn't work as expected.
